File: project-service/src/main/java/com/kapitonau/projectservice/model/audit/BaseAuditableModel.java
package com.kapitonau.projectservice.model.audit;

import jakarta.persistence.Column;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.MappedSuperclass;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.ZonedDateTime;

@MappedSuperclass
@EntityListeners(BaseAuditableModelListener.class)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public abstract class BaseAuditableModel {

    @Column(name = "created_time")
    @CreatedDate
    private ZonedDateTime createdTime;

    @Column(name = "modified_time")
    @LastModifiedDate
    private ZonedDateTime modifiedTime;

    @Column(name = "created_by")
    @CreatedBy
    private Long createdBy;

    @Column(name = "modified_by")
    @LastModifiedBy
    private Long modifiedBy;

}

File: task-service/src/main/kotlin/com/kapitonau/taskservice/model/audit/BaseAuditableModel.kt
package com.kapitonau.taskservice.model.audit

import jakarta.persistence.Column
import jakarta.persistence.MappedSuperclass
import org.springframework.data.annotation.CreatedBy
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import java.time.ZonedDateTime

@MappedSuperclass
abstract class BaseAuditableModel {

    @CreatedDate
    @Column(name = "created_time")
    open var createdTime: ZonedDateTime? = null

    @CreatedBy
    @Column(name = "created_by")
    open var createdBy: Long? = null

    @LastModifiedDate
    @Column(name = "modified_time")
    open var modifiedTime: ZonedDateTime? = null

    @LastModifiedDate
    @Column(name = "modified_by")
    open var modifiedBy: Long? = null

}